### PR TITLE
Updated template for roatp functions app

### PR DESCRIPTION
**Pull request by Knyatsine**

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -15,27 +19,58 @@
...
+            "defaultValue": []
         }
     },
     "variables": {
         "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
         "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
-        "resourceGroupName": "[concat(variables('resourceNamePrefix'), '-rg')]"
+        "resourceGroupName": "[concat(variables('resourceNamePrefix'), '-rg')]",
+        "appServicePlanName": "[concat(variables('resourceNamePrefix'),'-asp')]",
+        "funcAppName": "[concat(variables('resourceNamePrefix'),'fa')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:73]:
```suggestion
        "functionAppName": "[concat(variables('resourceNamePrefix'),'-fa')]"
```

File: azure/template.json [hunk 2]
```diff
@@ -46,12 +81,171 @@
...
+            "type": "Microsoft.Resources/deployments",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'),'application-insights.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "appInsightsName": {
+                        "value": "[variables('funcAppName')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:152]:
```suggestion
                        "value": "[variables('functionAppName')]"
```

File: azure/template.json [hunk 3]
```diff
@@ -46,12 +81,171 @@
...
+                    },
+                    "attachedService": {
+                        "value": "[variables('funcAppName')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:155]:
```suggestion
                        "value": "[variables('functionAppName')]"
```

File: azure/template.json [hunk 4]
```diff
@@ -46,12 +81,171 @@
...
+            "type": "Microsoft.Resources/deployments",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'),'function-app-v2.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "functionAppName": {
+                        "value": "[variables('funcAppName')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:173]:
```suggestion
                        "value": "[variables('functionAppName')]"
```

File: azure/template.json [hunk 5]
```diff
@@ -46,12 +81,171 @@
             "tags": "[parameters('tags')]",
             "properties": {
             }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2020-06-01",
+            "resourceGroup": "[variables('resourceGroupName')]",
+            "name": "worker-app-service-plan",
```
Comment (jack-cloud-platform) [inline azure/template.json:89]:
```suggestion
            "name": "function-app-app-service-plan",
```

Comment (jack-cloud-platform) [inline azure/template.json:89]:
```suggestion
            "name": "app-service-plan",
```

File: azure/template.json [hunk 6]
```diff
@@ -15,27 +19,58 @@
...
+            "defaultValue": []
         }
     },
     "variables": {
         "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
         "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
-        "resourceGroupName": "[concat(variables('resourceNamePrefix'), '-rg')]"
+        "resourceGroupName": "[concat(variables('resourceNamePrefix'), '-rg')]",
+        "appServicePlanName": "[concat(variables('resourceNamePrefix'),'-asp')]",
+        "functionAppName": "[concat(variables('resourceNamePrefix'),'fa')]"
```
Comment (jack-cloud-platform) [inline azure/template.json:73]:
```suggestion
        "functionAppName": "[concat(variables('resourceNamePrefix'),'-fa')]"
```

File: azure/template.json [hunk 7]
```diff
@@ -46,12 +81,171 @@
...
+                                {
+                                    "name": "AzureWebJobsStorage",
+                                    "value": "[parameters('sharedStorageAccountConnectionString')]"
+                                },
+                                {
+                                    "name": "ConfigurationStorageConnectionString",
+                                    "value": "[parameters('configurationStorageConnectionString')]"
+                                },
+                                {
+                                    "name": "Environment",
```
Comment (adam230594) [inline azure/template.json:207]: EnvironmentName would be best here

File: azure/template.json [hunk 8]
```diff
@@ -5,6 +5,10 @@
         "resourceEnvironmentName": {
             "type": "string"
         },
+        "version": {
```
Comment (adam230594) [inline azure/template.json:8]: I think this should just be hardcoded to 1.0 in the app settings to reduce the amount of parameters. Fine as it is though

File: azure/template.json [hunk 9]
```diff
@@ -15,27 +15,58 @@
             "type": "string",
             "defaultValue": "__SolutionBaseName__"
```
Comment (jack-cloud-platform) [inline azure/template.json:16]:
```suggestion
            "defaultValue": "SFA.DAS.Roatp.Functions"
```

**Comment by jack-cloud-platform** [comment]
Can you add dependsOn for the deployments that will deploy to the resource group that is being deployed? So release is successful first time

**Review by jack-cloud-platform — APPROVED** [review]

**Pull request merged**